### PR TITLE
Proposal: Send activeMembership data with all AI track events

### PR DESCRIPTION
**Pull request by stephenkao**
This changeset updates the AI track calls to always send `activeMembership` -- it should be applicable to `PageviewData`, `PageviewPerformanceData`, and `RemoteDependencyData` since those are active by default, but it should also apply to `EventData` and `MetricData` if we choose to include manually tracked events as well.  (I'll double-check this, though!)

A couple of questions:
- Are there any issues with sending this information on _every_ tracked event?
- Do we need to flatten this data so we only send primitive types?  Right now it's just a stringified object, but I don't have access to the AI dashboard so I don't know what our aggregation capabilities are😅

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Log in
2. Start navigating around
3. Check the network tab to see `track` requests and ensure that `activeMembership` is being sent along -- it should be under `baseData.data.properties.activeMembership`:

Screenshot 2022-12-08 at 11 11 16 AM

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: frontend-react/src/TelemetryService.ts [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
 import { ReactPlugin } from "@microsoft/applicationinsights-react-js";
 
+import { getSessionMembershipState } from "./utils/SessionStorageTools";
+
+import { appQueryClient } from "./network/QueryClients";
+
 let reactPlugin: ReactPlugin | null = null;
 let appInsights: ApplicationInsights | null = null;
 
+(window as any).appQueryClient = appQueryClient;
```
Comment (stephenkao) [inline frontend-react/src/TelemetryService.ts:11]: Oops, will remove :)

**Review by kevinhaube — COMMENTED** [review]
>Are there any issues with sending this information on every tracked event?

The first possible downside I could see is sending a bunch of "undefined"s for logged-out users. That said, not sure how it affects queries. We can tinker with it!

>Do we need to flatten this data so we only send primitive types? Right now it's just a stringified object, but I don't have access to the AI dashboard so I don't know what our aggregation capabilities are😅

It seemed like yesterday it was grouping numeric values and string values from these objects accordingly! I think it stringifies it and then parses it again on the Azure side. We'll check!

**Comment by stephenkao** [comment]
> > Are there any issues with sending this information on every tracked event?
> 
> The first possible downside I could see is sending a bunch of "undefined"s for logged-out users. That said, not sure how it affects queries. We can tinker with it!

I think that's fine since we can't really send anything else for logged-out users.  We'll at least have the AI-generated session id for logged-out users, but there's nothing else available to us if the user isn't logged in! 🤔 

> > Do we need to flatten this data so we only send primitive types? Right now it's just a stringified object, but I don't have access to the AI dashboard so I don't know what our aggregation capabilities are😅
> 
> It seemed like yesterday it was grouping numeric values and string values from these objects accordingly! I think it stringifies it and then parses it again on the Azure side. We'll check!

Yep!  My question was about whether having a stringified object here would make it harder for us to aggregate on the AI dashboard side (like if we'd run into issues with whatever the equivalent of `GROUP BY` is on the dashboard).  I think I remember seeing the properties get properly parsed from the stringified JSON, but would you mind double-checking if you could do something like group by `properties.activeMembership.parsedName` on the AI dashboard?  (Might be [the `summarize` operator](https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/summarizeoperator)?)

**Comment by stephenkao** [comment]
Confirmed that this works with the `useTrackEvent` hook out of the box as well, so I can close https://github.com/CDCgov/prime-reportstream/pull/7608 if we choose to go forward with this changeset!

Screenshot 2022-12-08 at 1 50 23 PM

**Comment by stephenkao** [comment]
@kevinhaube Now that we've verified that nested fields in the stringified JSON can work, can I get an approval on this?  :P  Otherwise, is there anything else we need here?

**Review by kevinhaube — APPROVED** [review]
🥳

**Pull request merged**